validator/client/validation_client.go

```diff
@@ -67,22 +68,21 @@ func (c *ValidationClient) Start(ctx context.Context) error {
 	if len(name) == 0 {
 		return errors.New("couldn't read name from server")
 	}
-	var stylusArchs []string
+	var stylusArchs []rawdb.Target
 	if err := c.client.CallContext(ctx, &stylusArchs, server_api.Namespace+"_stylusArchs"); err != nil {
 		var rpcError rpc.Error
 		ok := errors.As(err, &rpcError)
 		if !ok || rpcError.ErrorCode() != -32601 {
 			return fmt.Errorf("could not read stylus arch from server: %w", err)
 		}
-		stylusArchs = []string{"pre-stylus"} // validation does not support stylus
-	} else {
-		if len(stylusArchs) == 0 {
-			return fmt.Errorf("could not read stylus archs from validation server")
-		}
-		for _, stylusArch := range stylusArchs {
-			if stylusArch != "wavm" && stylusArch != runtime.GOARCH && stylusArch != "mock" {
-				return fmt.Errorf("unsupported stylus architecture: %v", stylusArch)
-			}
+		stylusArchs = []rawdb.Target{"pre-stylus"} // validation does not support stylus
+	}
+	if len(stylusArchs) == 0 {
+		return fmt.Errorf("could not read stylus archs from validation server")
+	}
+	for _, stylusArch := range stylusArchs {
+		if stylusArch != rawdb.TargetWavm && stylusArch != programs.LocalTarget() && stylusArch != "mock" {
+			return fmt.Errorf("unsupported stylus architecture: %v", stylusArch)
```

Comment:
I missed that case. I am not sure if passing further `pre-stylus` arch solves the issue fully -`validationEntry.ToInput` will fail if stylus program call was recorded (what could be desired), but also will include UserWasms with empty map under "pre-stylus" key

Reply:
I changed the logic to return empty stylusArchs for clients that does not support stylus and added a check in `validationEntry.ToInput` to fail if stylusArchs is empty but some stylus program was recorded. Let me know what you think about this approach

Reply:
I like having a stylusArchs of "pre-stylus" because it will error early and in a pretty clear way if we try to use it to validate a block with a stylus contract, where an empty stylus-archs will create errors that are quite hard to debug.
I think either accept "pre-stylus" as an option here, or move that entire loop into "else" of the above if so it's only used if there id no error and len is not 0